activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -771,6 +771,10 @@ def can_perform_case_insensitive_comparison_for?(column)
             sql = <<-end_sql
               SELECT exists(
                 SELECT * FROM pg_proc
+                WHERE proname = 'lower'
+                  AND proargtypes = '#{column.sql_type}'::regtype::oid::text::oidvector
+              ) OR exists(
+                SELECT * FROM pg_proc
                 INNER JOIN pg_cast
                   ON casttarget::text::oidvector = proargtypes
                 WHERE proname = 'lower'
```

Comment:
.. and, one the line below this one: `#{quote column.sql_type}::regtype` (no need for `::oid`)